projects/plugins/jetpack/class.jetpack.php

```diff
@@ -1614,13 +1614,19 @@ public static function is_development_mode() {
 	 * A site is considered as being onboarded if it currently has an onboarding token.
 	 *
 	 * @since 5.8
+	 * @deprecated Use \Automattic\Jetpack\Status()->is_onboarding()
 	 *
 	 * @access public
 	 * @static
 	 *
 	 * @return bool True if the site is currently onboarding, false otherwise
 	 */
 	public static function is_onboarding() {
+		_deprecated_function( __METHOD__, 'jetpack-10.9', 'Automattic\\Jetpack\\Status\\is_onboarding' );
```

Comment:
The only problem here is that this seems to get called on every wp-admin pageview when the site's not connected :(, so if they are logging notices, we'll be filling up their logs very quickly.

Reply:
So maybe we move the method_exists check down to the `load_modules()`, and only reference the deprecated static class if it doesn't exist.

Reply:
Yeah, good catch, thank you! I did what you said, but I decided to leave the check in both places, so that we can call the deprecated method everywhere and still get that deprecation notice.
As for the load_modules path - we still have to call the old method if the new isn't available, I doubt we can do anything more for those people that will have notices overflowing short of removing that notice entirely.

Reply:
> I doubt we can do anything more for those people that will have notices overflowing short of removing that notice entirely

True, but with this approach it should lead to hitting the deprecated method much less frequently